### PR TITLE
vacuum: port to mkDerivation, add zlib for hydra failure

### PR DESCRIPTION
**Pull request by grahamc**
###### Things done:
- [X] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by grahamc** [comment]
re  #4210

File: pkgs/applications/networking/instant-messengers/vacuum/default.nix [hunk 1]
```diff
@@ -1,56 +1,30 @@
-x@{builderDefsPackage
+{ stdenv, lib, fetchurl
   , qt4, openssl
   , xproto, libX11, libXScrnSaver, scrnsaverproto
-  , xz
+  , xz, zlib
   , ...}:
```
Comment (joachifm) [inline pkgs/applications/networking/instant-messengers/vacuum/default.nix]: The rest parameter can be removed

File: pkgs/applications/networking/instant-messengers/vacuum/default.nix [hunk 2]
```diff
@@ -1,56 +1,30 @@
...
 
-  buildInputs = map (n: builtins.getAttr n x)
-    (builtins.attrNames (builtins.removeAttrs x helperArgNames));
-  sourceInfo = rec {
-    version="1.2.4";
-    baseName="vacuum-im";
-    name="${baseName}-${version}";
-    url="https://googledrive.com/host/0B7A5K_290X8-d1hjQmJaSGZmTTA/vacuum-1.2.4.tar.gz";
+  src = fetchurl {
+    url="https://googledrive.com/host/0B7A5K_290X8-d1hjQmJaSGZmTTA/vacuum-${version}.tar.gz";
```
Comment (joachifm) [inline pkgs/applications/networking/instant-messengers/vacuum/default.nix]: Please add whitespaces around the equals sign.

Comment (grahamc) [inline pkgs/applications/networking/instant-messengers/vacuum/default.nix]: fixed

File: pkgs/applications/networking/instant-messengers/vacuum/default.nix (hunk 1 above)
Comment (grahamc) [inline pkgs/applications/networking/instant-messengers/vacuum/default.nix]: fixed

**Comment by grahamc** [comment]
I think this is broken on 16.03, so should get backported.

**Comment by joachifm** [comment]
The build is still broken the release branch: https://hydra.nixos.org/build/33612457

**Pull request merged**